### PR TITLE
added unit test for density kernel

### PR DESCRIPTION
**Pull request by sekelle**

### CODE REVIEW

**Review by jgphpc — CHANGES_REQUESTED** [review]
temporary blocking merge to give me some time to validate/debug the ci (I will unlock when ready).

File: include/sph/test/kernel/density.cpp
Comment (rmcabezon) [inline include/sph/test/kernel/density.cpp:66]: With these distances and h(0)=5, all particles will be neighbors. I'd recommend using a smaller h so that some particles are neighbors and others are not. In that way, you ensure also that the kernel is adding a zero when a particle is qualified as a neighbor but in fact, is not. This might happen if we define the neighborhood with a gather models instead of a scatter model.

File: include/sph/test/kernel/density.cpp
Comment (rmcabezon) [inline include/sph/test/kernel/density.cpp:111]: Same comment as above. Either reduce h(0) or add some particles farther than 2h(0).

**Review by rmcabezon — CHANGES_REQUESTED** [review]

File: include/sph/test/kernel/density.cpp
Comment (sekelle) [reply include/sph/test/kernel/density.cpp:66]: The current implementation of the density kernel assumes that the neighbor list is correct. Whatever is listed as a neighbor will be added, even if the distance turns out to be bigger than 2*h. So I can't really test correct neighbor search here.

**Comment by sekelle** [comment]
@jgphpc This MR need not be held up just because the test is not in the CI yet. You can add it to the CI in a separate MR.

**Review by jgphpc — APPROVED** [review]
[  PASSED  ] Ran 7/8 test case(s) from 8 check(s) (0 failure(s), 1 skipped)
Will add new unittest after merging.

**Review by rmcabezon — APPROVED** [review]

**Comment by jgphpc** [comment]
Check gitlab tests results before merging: [  PASSED  ] Ran 31/32 test case(s) from 32 check(s) (0 failure(s), 1 skipped) https://git.scicore.unibas.ch/j.piccinali/mirror_test/-/pipelines/13401 github_pr=105
fa8c245

**Pull request merged**
